osu.Game/Rulesets/Scoring/JudgementProcessor.cs

```diff
@@ -107,6 +108,17 @@ protected virtual void Reset(bool storeResults)
             JudgedHits = 0;
         }
 
+        public virtual void ResetFromReplayFrame(Ruleset ruleset, ReplayFrame frame)
```

Comment:
Kinda weird the ruleset is passed in here, but seems best for now (there's a lot of usages of this class, and many which aren't abstracted to a ruleset level to allow them to ctor populate correctly).